src/module_2/eda_groceries.md

```diff
@@ -0,0 +1,506 @@
+    dtype: object
+    
+    
+    df_abandoned:
+    Forma: (5457, 4)
+    Tamaño: 21828
+    Clases de datos (columnas y tipos):
+    id                     int64
+    user_id               object
+    created_at    datetime64[us]
+    variant_id            object
+    dtype: object
+    
+    
+    df_inventory:
+    Forma: (1733, 6)
+    Tamaño: 10398
+    Clases de datos (columnas y tipos):
+    variant_id            int64
+    price               float64
+    compare_at_price    float64
+    vendor               object
+    product_type         object
+    tags                 object
+    dtype: object
+    
+    
+    df_users:
+    Forma: (4983, 10)
+    Tamaño: 49830
+    Clases de datos (columnas y tipos):
+    user_id                   object
+    user_segment              object
+    user_nuts1                object
+    first_ordered_at          object
+    customer_cohort_month     object
+    count_people             float64
+    count_adults             float64
+    count_children           float64
+    count_babies             float64
+    count_pets               float64
+    dtype: object
+    
+    
+
+
+**Thinking of the supermarket as my business, I have oriented the EDA towards those points that add value when making decisions that increase revenue and the number of sales.**
+
+1. Customer study:
+ - Where do I have the most impact?: In locations with less impact, I can enhance advertising efforts.
+
+
+```python
+# df_users.info()
+# df_users['user_nuts1'] -> 4932 non-null/4986. Very small amount, I am dropping them to avoid imputing values I cannot verify
+df_users["user_nuts1"].dropna().value_counts(normalize=True).plot(
+    kind="bar", figsize=(15, 5)
+)
+```
+
+
+
+
+    <Axes: xlabel='user_nuts1'>
+
+
+
+
+    
+![png](eda_groceries_files/eda_groceries_10_1.png)
+    
+
+
+2- Analysis of revenue trends:
+
+- Distribution of 'order' amounts (a customer's purchase): This will reveal the spread and most common purchase values from customers. Ideally, there would be a higher concentration of 'orders' with larger amounts. This will help understand the current state and serve as a KPI to assess the effectiveness of strategies aimed at increasing the amount each customer spends per purchase.
+- Distribution of the prices of available products: This will provide insight into the economic weight of the products available to customers.
+- Evolution of total revenue over time: This information will help quantify the financial impact of new measures."
+
+
+```python
+variant_price = (
+    df_orders.explode("variant_id")
+    .reset_index(drop=True)
+    .merge(df_inventory.loc[:, ["variant_id", "price"]], on="variant_id", how="left")
+)
+# variant_price -> [id, user_id, created_at, variant_id, price]
+order_revenue = variant_price.groupby("id")["price"].sum().reset_index()
+print(order_revenue.describe())
+order_revenue["price"].hist(bins=100, log=True)
+plt.xlabel("Total Revenue per Order")
+plt.ylabel("Frequency")
```

Comment:
Even if we repeat ourselves with this. Titles in graphs are always super useful to read info at a quick glance. Especially if later we want to use them in a report or dashboard separated from our code.